src/twisted/application/twist/test/test_twist.py

```diff
@@ -191,3 +191,80 @@ def run(self):
             )
         )
         self.assertEqual(runners[0].runs, 1)
+
+
+
+class TwistExitTests(twisted.trial.unittest.TestCase):
+
+    """
+    Tests to verify that the Twist script takes the expected actions related
+    to signals and the reactor.
+    """
+
+    def setUp(self):
+        self.exitWithSignalCalled = False
+
+        def fakeExitWithSignal(sig):
+            """
+            Fake to capture whether L{twisted.application._exitWithSignal
+            was called.
+
+            @param sig: Signal value
+            @type sig: C{int}
+            """
+            self.exitWithSignalCalled = True
+
+        self.patch(_twist, '_exitWithSignal', fakeExitWithSignal)
+
+        def startLogging(_):
+            """
+            Prevent Runner from adding new log observers or other
+            tests outside this module will fail.
+
+            @param _: Unused self param
+            """
+            pass
+
+        self.patch(Runner, 'startLogging', startLogging)
```

Comment:
@glyph I've looked in to this a bit, and there aren't many good options with the existing code.  Runner is imported into _twist, it isn't an option that can be easily injected.  The default runner can be used, but somewhere the call to `globalLogBegginer.beginLoggingTo` needs to be prevented or downstream tests fail.

WIth the existing code, the following seem like the options:
1.  Leave it as is.
2.  Patch the Runner itself, as is done in other tests.
3.  Patch globalLogBeginner in twisted.application.runner._runner with a new LogBeginner for each test.
4.  Patch 'beginLoggingTo' on globalLogBeginner as is done in test_twistd.

To inject a runner, the runner should probably be sourced from TwistOptions, rather than imported globally, and be selectable.  While I am more than willing to do this change, it does feel like it is significant enough to warrant its own ticket.  Specfiically, it seems like a fairly heavy side effect for an unrelated change.

Thoughts?